node/src/components/contract_runtime.rs

```diff
@@ -101,16 +101,18 @@ impl From<&BlockHeader> for ExecutionPreState {
     }
 }
 
+type ExecQueue = Arc<Mutex<BTreeMap<u64, (FinalizedBlock, Vec<Deploy>)>>>;
```

Comment:
As discussed, I suggest we move this queue to `LinearChain`.

Moreover, it's not necessary for it to have its deploys, we can just get those from storage.

Reply:
`LinearChain` deals with blocks that are already in ... linear chain. A `FinalizedBlock` is not yet a linear chain block - it's in a different bounded context, it's been only finalized by consensus and has to be executed and then transformed into a linear chain block. This queue is in my opinion in a proper place - we used to have a `BlockExecutor` object where that queue would fit more but it has been merged with `ContractRuntime`.

Reply:
You are right. I agree we should not couple anything unnecessary to the `LinearChain` component.  The issue with this queue is that it is not used in the `ContractRuntime` component when we are fast-syncing.

@dwerner and I have discussed this and we are thinking now it should become task-local state (in fast-sync the `ExecutionPreState` is similarly task-local).  Planning on doing this in follow-up work.